src/haz3lcore/dynamics/transition/Unboxing.re

```diff
@@ -93,27 +88,12 @@ let rec unbox: type a. (unbox_request(a), DHExp.t) => unboxed(a) =
       } else {
         DoesNotMatch;
       }
-    | (
-        TupLabel(tl),
-        Cast(t, {term: TupLabel(_, ty1), _}, {term: TupLabel(_, ty2), _}),
-      ) =>
-      let* t = unbox(TupLabel(tl), t);
-      let t = fixup_cast(Cast(t, ty1, ty2) |> DHExp.fresh);
-      Matches(t);
     | (TupLabel(_), _) => Matches(expr)
-
-    /* Remove Tuplabels from casts otherwise */
-    | (_, Cast(e, {term: TupLabel(_, e1), _}, e2)) =>
-      switch (DHExp.term_of(e)) {
-      | TupLabel(_, e) => unbox(request, Cast(e, e1, e2) |> DHExp.fresh)
-      | _ => unbox(request, Cast(e, e1, e2) |> DHExp.fresh)
-      }
-
     /* Base types are always already unboxed because of the ITCastID rule*/
     | (Atom(r), Atom(x)) =>
       switch (Atom.unbox(r, x)) {
       | Some(x) => Matches(x)
-      | None => DoesNotMatch
+      | None => IndetMatch
```

Comment:
@Negabinary This seems more correct to me even absent the tdos approach. Not sure if you have thoughts